File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = [0] * len(prices)
    stack = []
    
    for current_time in range(len(prices)):
        while stack and prices[current_time] < prices[stack[-1]]:
            prev_time = stack.pop()
            answer[prev_time] = current_time - prev_time
            

        stack.append(current_time)
    last_time = len(prices) - 1
    
    while stack:
        prev_time = stack.pop()
        answer[prev_time] = last_time - prev_time
    
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
  const answer = Array(prices.length).fill(0);
  const stack = [];

  for (let currentTime = 0; currentTime < prices.length; currentTime++) {
    while (
      stack.length &&
      prices[currentTime] < prices[stack[stack.length - 1]]
    ) {
      const prevTime = stack.pop();
      answer[prevTime] = currentTime - prevTime;
    }

    stack.push(currentTime);
  }

  const lastTime = prices.length - 1;
  while (stack.length) {
    const prevTime = stack.pop();
    answer[prevTime] = lastTime - prevTime;
  }

  return answer;
}